src/GafferSceneUIModule/ToolBinding.cpp

```diff
@@ -183,12 +183,57 @@ object acquireTransformEdit( const TransformTool::Selection &s, bool createIfNec
 	return p ? object( *p ) : object();
 }
 
+struct SelectModeHelper
+{
+	SelectModeHelper( object fn ) : m_fn( fn )
+	{
+	}
+
+	GafferScene::ScenePlug::ScenePath operator()(
+		const GafferScene::ScenePlug *scene,
+		const GafferScene::ScenePlug::ScenePath &path
+	)
+	{
+		IECorePython::ScopedGILLock gilLock;
+
+		try
+		{
+			const std::string pathString = GafferScene::ScenePlug::pathToString( path );
+			GafferScene::ScenePlug::ScenePath newPath = extract<GafferScene::ScenePlug::ScenePath>(
+				m_fn( GafferScene::ScenePlugPtr( const_cast<GafferScene::ScenePlug *>( scene ) ), pathString )
+			);
+			return newPath;
+		}
+		catch( const boost::python::error_already_set & )
+		{
+			ExceptionAlgo::translatePythonException();
+		}
+	}
+	private :
+		object m_fn;
```

Comment:
I think we'll get away with this because it's unlikely that anything is registered or deregistered from pure C++, but I'll mention it anyway : the destructor for `SelectModeHelper` will destroy this Python object without acquiring the GIL first. That has caused crashes in registries that are modified from pure C++. An alternative approach is the one I took in the Renderer registry recently : https://github.com/GafferHQ/gaffer/blob/main/src/GafferSceneModule/RenderBinding.cpp#L121-L142. The lambda there also cuts down on the amount of boilerplate a little bit. Might be worth considering...

Reply:
That looks cleaner to me, and worth fixing before it becomes a surprise problem later on. Fixed in https://github.com/GafferHQ/gaffer/pull/5720/commits/342035f8dfdc6dc7d5b683e025d388f68f12a6a3